tests/foreman/cli/test_computeresource_osp.py

```diff
@@ -0,0 +1,274 @@
+        compute_resource = make_compute_resource({
+            u'name': name,
+            u'provider': 'Openstack',
+            u'user': self.username,
+            u'password': self.password,
+            u'tenant': self.tenant,
+            u'url': self.current_osp_url
+        })
+        self.assertEquals(compute_resource['name'], name)
+        with self.assertRaises(CLIFactoryError):
+            make_compute_resource({
+                u'name': name,
+                u'provider': 'Openstack',
+                u'user': self.username,
+                u'password': self.password,
+                u'tenant': self.tenant,
+                u'url': self.current_osp_url
+            })
+
+    @tier1
+    @run_only_on('sat')
+    def test_positive_update_name(self):
+        """Openstack compute resource positive update
+
+        :id: 16eb2def-34d5-49c5-be22-88139fef7f97
+
+        :steps:
+
+            1. Create a osp compute resource
+            2. Update the name of the created compute resource
+
+        :expectedresults: Compute Resource is successfully updated
+
+        :CaseImportance: Critical
+        """
+        new_name = gen_string('alpha')
+        comp_res = make_compute_resource({
+            u'provider': 'Openstack',
+            u'user': self.username,
+            u'password': self.password,
+            u'tenant': self.tenant,
+            u'url': self.current_osp_url
+        })
+        self.assertTrue(comp_res['name'])
+        ComputeResource.update({
+            'name': comp_res['name'],
+            'new-name': new_name
+        })
+        self.assertEqual(
+            new_name,
+            ComputeResource.info({'id': comp_res['id']})['name']
+        )
+
+    @tier3
+    @run_only_on('sat')
+    @stubbed()
+    def test_positive_provision_osp_with_host_group(self):
+        """Provision a host on Openstack compute resource with
+        the help of hostgroup.
+
+        :id: d85df090-875f-4f2b-b0f2-708efd5f50f3
+
+        :setup: Hostgroup and provisioning setup like domain, subnet etc.
+
+        :steps:
+
+            1. Create a osp compute resource.
+            2. Create a host on osp compute resource using the Hostgroup
+            3. Use compute-attributes parameter to specify key-value parameters
+               regarding the virtual machine.
+            4. Provision the host.
+
+        :expectedresults: The host should be provisioned with host group
+
+        :caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    @upgrade
+    def test_positive_provision_osp_without_host_group(self):
+        """Provision a host on Openstack compute resource without
+        the help of hostgroup.
+
+        :id: 644000fc-a131-4df1-89ad-897f0c741f06
+
+        :setup: Provisioning setup like domain, subnet etc.
+
+        :steps:
+
+            1. Create a osp compute resource.
+            2. Create a host on osp compute resource.
```

Comment:
Can we create the host before provisioning?
or the host is created when provisioning is successfully finished?